### PR TITLE
Pass X and Y Scale factors through to rasterio.warp.project.

### LINKED ISSUE #1448: Distorted data along fractional Dask chunks when loading data with `dc.load` and Dask
### Expected behaviour
When loading data with Open Data Cube, we expect to be able to use Dask to efficiently parallelise workflows and reduce peak memory usage. Part of using Dask is an implicit assumption that data loaded by Dask will be identical to data loaded without Dask.

### Actual behaviour
There appears to be a bug in `datacube.load` that is causing data loaded with Dask to be distorted along edges of Dask chunks. We first noticed data being blurred/stretched along the right side of our data loads when loaded with Dask - see animation below which shows the effect along the right side of the image:
 
dask_vs_nodask

Looking into this deeper, it appears that if a non-"nearest" resampling method is used (e.g. "cubic", "bilinear") and a Dask chunk extends off the right side of a data load, the data from that chunk will be strongly distorted. For example, if `dask_chunks={"time": 1, "x": 2048, "y": 2048}` and our data load is 2100 pixels wide, the right-most 52 pixels will be affected. If Dask is not used, `dc.load` works perfectly. 

### Steps to reproduce the behaviour
We've put together a Jupyter Notebook with a reproducible example here - it includes several examples of the issue, including some visualisations of how much this affects loaded data: https://gist.github.com/robbibt/4d6905d5d13137bb55362c4a6f1c18b8

For example, here is the difference between data loaded with and without Dask - note the band of strong differences along the right-hand side of the image:
image

### Environment information

* Which ``datacube --version`` are you using? 1.8.12
* What datacube deployment/enviornment are you running against?
Prod DEA Sandbox

### PR DESCRIPTION
**Pull request by SpacemanPaul**
### Reason for this pull request

Attempted fix for issue #1448

### Proposed changes

- Pass integer scale factors through to rasterio.warp.project (as XSCALE and YSCALE parameters).

 - [x] Closes #1448
 - [x] Tests added / passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

----
:books: Documentation preview :books:: https://datacube-core--1450.org.readthedocs.build/en/1450/

### CODE REVIEW

**Review by robbibt — APPROVED** [review]
Can confirm this resolves our issue! I've re-run all my previous analyses, and there's no longer any trace of interaction between resampling method and data load extent/aspect ratio/size. For example, in #1448 we generated a graph showing the difference between full and fractional loads by resampling method:

Before:
image

After:
image

As far as I can tell, resampling should now behave the same regardless of the shape and size AOI requested by the user. This means that that we can rely on using Dask with spatial chunking, which is fantastic.

Thanks very much for this fix!

File: datacube/storage/_read.py [hunk 1]
```diff
@@ -129,6 +129,7 @@ def read_time_slice(rdr,
 
     is_nn = is_resampling_nn(resampling)
     scale = pick_read_scale(rr.scale, rdr)
+    scale_xy = [pick_read_scale(s, rdr) for s in rr.scale2]
```
Comment (Kirill888) [inline datacube/storage/_read.py]: Not sure if `pick_read_scale` is needed here, certainly no need to pass through `rdr` object, right now it's no-op regardless but the idea is to pick an overview image best fitting current scale.
```suggestion
    xscale, yscale = [pick_read_scale(s) for s in rr.scale2]
```

File: datacube/storage/_read.py [hunk 2]
```diff
@@ -188,7 +189,8 @@ def norm_read_args(roi, shape, extra_dim_index):
                         src_nodata=rdr.nodata, dst_nodata=dst_nodata)
         else:
             rio_reproject(pix, dst, src_gbox, dst_gbox, resampling,
-                          src_nodata=rdr.nodata, dst_nodata=dst_nodata)
+                          src_nodata=rdr.nodata, dst_nodata=dst_nodata,
+                          XSCALE=scale_xy[0], YSCALE=scale_xy[1])
```
Comment (Kirill888) [inline datacube/storage/_read.py]: prefer use de-structure over indexing:

```suggestion
                          XSCALE=xscale, YSCALE=yscale)
```

main reason is that errors are easier to locate that way

File: datacube/storage/_read.py (hunk 1 above)
Comment (SpacemanPaul) [reply datacube/storage/_read.py]: Yeah, it's mostly just force it to be an integer.

**Pull request merged**